python/plugins/processing/script/ScriptTemplate.py

```diff
@@ -190,7 +190,15 @@ def processAlgorithm(self, parameters, context, feedback):
                 'MITER_LIMIT': 2,
                 'DISSOLVE': False,
                 'OUTPUT': 'memory:'
-            }, context=context, feedback=feedback)['OUTPUT']
+            }, context=context, feedback=feedback, is_child_algorithm=True)['OUTPUT']
+            
+            # For child algorithms, we disable to default post-processing step where
+            # layer ownership is transferred from the context to the caller. In this
+            # case, we NEED the ownership to remain with the context, so that further
+            # steps in the algorithm have guaranteed access to the layer.
+            # To hand ownership over to the context, use:
```

Comment:
```suggestion
            # If we know for certain that no other parts of the algorithm (or models containing this algorithm) will require the layer, then we can take ownership directly by calling context.takeResultLayer. This causes the layer's ownership to be transferred to Python, so that the normal Python variable scope and garbage collection rules apply.
```